x-pack/plugins/data_visualizer/public/application/index_data_visualizer/embeddables/field_stats/field_stats_factory.tsx

```diff
@@ -0,0 +1,327 @@
+      };
+      const {
+        api: timeRangeApi,
+        comparators: timeRangeComparators,
+        serialize: serializeTimeRange,
+      } = initializeTimeRange(state);
+
+      const { titlesApi, titleComparators, serializeTitles } = initializeTitles(state);
+
+      const {
+        fieldStatsControlsApi,
+        dataLoadingApi,
+        fieldStatsControlsComparators,
+        serializeFieldStatsChartState,
+        onFieldStatsTableDestroy,
+        resetData$,
+      } = initializeFieldStatsControls(state);
+
+      const defaultDataViewId = await deps.data.dataViews.getDefaultId();
+      const dataViews$ = new BehaviorSubject<DataView[] | undefined>(
+        state.dataViewId !== '' && defaultDataViewId
+          ? [await deps.data.dataViews.get(state.dataViewId ?? defaultDataViewId)]
+          : undefined
+      );
+
+      const subscriptions = new Subscription();
+      subscriptions.add(
+        fieldStatsControlsApi.dataViewId$
+          .pipe(
+            skip(1),
+            skipWhile((dataViewId) => dataViewId === '' && defaultDataViewId === null),
+            switchMap((dataViewId) => deps.data.dataViews.get(dataViewId ?? defaultDataViewId))
+          )
+          .subscribe((nextSelectedDataView) => {
+            dataViews$.next([nextSelectedDataView]);
+          })
+      );
+
+      const api = buildApi(
+        {
+          ...timeRangeApi,
+          ...titlesApi,
+          ...fieldStatsControlsApi,
+          ...dataLoadingApi,
+          getTypeDisplayName: () =>
+            i18n.translate('xpack.dataVisualizer.fieldStats.typeDisplayName', {
+              defaultMessage: 'field statistics',
+            }),
+          isEditingEnabled: () => true,
+          onEdit: async () => {
+            try {
+              const { resolveEmbeddableFieldStatsUserInput } = await import(
+                './resolve_field_stats_embeddable_input'
+              );
+              const chartState = serializeFieldStatsChartState();
+              const nextUpdate = await resolveEmbeddableFieldStatsUserInput(
+                coreStart,
+                pluginStart,
+                parentApi,
+                uuid,
+                chartState,
+                fieldStatsControlsApi
+              );
+
+              fieldStatsControlsApi.updateUserInput(nextUpdate);
+            } catch (e) {
+              return Promise.reject(e);
+            }
+          },
+          dataViews: dataViews$,
+          serializeState: () => {
+            const dataViewId = fieldStatsControlsApi.dataViewId$?.getValue();
+            const references: Reference[] = dataViewId
+              ? [
+                  {
+                    type: DATA_VIEW_SAVED_OBJECT_TYPE,
+                    name: FIELD_STATS_DATA_VIEW_REF_NAME,
+                    id: dataViewId,
+                  },
+                ]
+              : [];
+            return {
+              rawState: {
+                timeRange: undefined,
```

Comment:
Why are you setting `timeRange: undefined` when `...serializeTimeRange()` will set timeRange. You should be able to remove this line

Reply:
Updated here https://github.com/elastic/kibana/pull/184030/commits/f990d2278cc309c21b0893cd076b8e2f2ec127b1